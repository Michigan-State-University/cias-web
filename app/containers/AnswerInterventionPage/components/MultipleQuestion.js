import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Column from 'components/Column';
import Row from 'components/Row';
import Img from 'components/Img';
import Question from 'models/Intervention/Question';
import HoverableBox from 'components/Box/HoverableBox';
import checkbox from 'assets/svg/checkbox.svg';
import checkboxChecked from 'assets/svg/checkbox-checked.svg';

const margin = 21;

const MultipleQuestion = ({ question, answerBody, selectAnswer }) => {
  const [selectedAnswersIndex, setSelectedAnswersIndex] = useState([]);

  const check = (payload, value, name, index) => {
    const selectedAnswer = {
      variable: name,
      payload,
      value,
    };
    if (selectedAnswersIndex.includes(index)) {
      setSelectedAnswersIndex(
        selectedAnswersIndex.filter(item => item !== index),
      );
      selectAnswer(
        answerBody.filter(item => item.value !== selectedAnswer.value),
      );
    } else {
      setSelectedAnswersIndex([...selectedAnswersIndex, index]);
      selectAnswer([...answerBody, selectedAnswer]);
    }
  };

  const {
    body: { data },
  } = question;
  return (
    <Column mt={10} mb={10}>
      {data.map((questionAnswer, index) => {
        const {
          payload,
          variable: { name, value },
        } = questionAnswer;
        return (
          <Row key={`question-${question.id}-el-${index}`}>
            <HoverableBox
              px={margin}
              py={14}
              mx={margin}
              width={`calc(100% + ${margin}px)`}
              clickable
              onClick={() => check(payload, value, name, index)}
            >
              <Row align="center">
                <Img
                  src={
                    selectedAnswersIndex.includes(index)
                      ? checkboxChecked
                      : checkbox
                  }
                  mr={16}
                />
                {payload}
              </Row>
            </HoverableBox>
          </Row>
        );
      })}
    </Column>
  );
};

MultipleQuestion.propTypes = {
  question: PropTypes.shape(Question).isRequired,
  answerBody: PropTypes.any,
  selectAnswer: PropTypes.func,
};

export default MultipleQuestion;
