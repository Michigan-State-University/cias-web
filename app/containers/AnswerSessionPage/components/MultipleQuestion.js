import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import MultipleQuestionLayout from '../layouts/MultipleQuestionLayout';

const MultipleQuestion = ({
  question,
  answerBody,
  selectAnswer,
  isMobile,
  selectingAnswersDisabled,
}) => {
  const [selectedAnswersIndex, setSelectedAnswersIndex] = useState([]);

  const {
    body: { data },
    id,
  } = question;

  useEffect(() => {
    setSelectedAnswersIndex(
      answerBody.length ? answerBody.map((answer) => answer.index) : [],
    );
  }, [id]);

  const check = (value, name, index) => {
    const selectedAnswer = {
      var: name,
      value,
      index,
    };
    if (selectedAnswersIndex.includes(index)) {
      setSelectedAnswersIndex(
        selectedAnswersIndex.filter((item) => item !== index),
      );
      selectAnswer(
        // if unchecked answer has empty variable name, it will uncheck all answers without variable name
        // but these answers wouldn't be considered in formulas anyway (because they don't have any variable assigned)
        answerBody.filter((item) => item.var !== selectedAnswer.var),
      );
    } else {
      setSelectedAnswersIndex([...selectedAnswersIndex, index]);
      selectAnswer([...answerBody, selectedAnswer]);
    }
  };
  return (
    <MultipleQuestionLayout
      data={data}
      questionId={id}
      check={check}
      selectedAnswersIndex={selectedAnswersIndex}
      isMobile={isMobile}
      disabled={selectingAnswersDisabled}
    />
  );
};

MultipleQuestion.propTypes = {
  question: PropTypes.object.isRequired,
  answerBody: PropTypes.any,
  selectAnswer: PropTypes.func,
  isMobile: PropTypes.bool,
  selectingAnswersDisabled: PropTypes.bool,
};

export default MultipleQuestion;
