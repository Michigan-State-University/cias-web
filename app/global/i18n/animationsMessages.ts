import { defineMessages } from 'react-intl';

import { NarratorAnimation } from 'models/Narrator';

export const scope = 'app.global.Animations';

export default defineMessages<NarratorAnimation>({
  uncertain: {
    id: `${scope}.uncertain`,
    defaultMessage: 'Uncertain',
  },
  wave: {
    id: `${scope}.wave`,
    defaultMessage: 'Wave',
  },
  greet: {
    id: `${scope}.greet`,
    defaultMessage: 'Greet',
  },
  otohLeftHand: {
    id: `${scope}.otohLeftHand`,
    defaultMessage: 'On the other hand (left)',
  },
  otohRightHand: {
    id: `${scope}.otohRightHand`,
    defaultMessage: 'On the other hand (right)',
  },
  pointRight: {
    id: `${scope}.pointRight`,
    defaultMessage: 'Point right',
  },
  pointLeft: {
    id: `${scope}.pointLeft`,
    defaultMessage: 'Point left',
  },
  pointUp: {
    id: `${scope}.pointUp`,
    defaultMessage: 'Point up',
  },
  pointDown: {
    id: `${scope}.pointDown`,
    defaultMessage: 'Point down',
  },
  pointLeftUp: {
    id: `${scope}.pointLeftUp`,
    defaultMessage: 'Point left up',
  },
  pointRightUp: {
    id: `${scope}.pointRightUp`,
    defaultMessage: 'Point right up',
  },
  explain: {
    id: `${scope}.explain`,
    defaultMessage: 'Explain',
  },
  rest: {
    id: `${scope}.rest`,
    defaultMessage: 'Rest',
  },
  restShoulderRubbing: {
    id: `${scope}.restShoulderRubbing`,
    defaultMessage: 'Rest shoulder rubbing',
  },
  restStretching: {
    id: `${scope}.restStretching`,
    defaultMessage: 'Rest stretching',
  },
  restWeightShift: {
    id: `${scope}.restWeightShift`,
    defaultMessage: 'Rest weight shift',
  },
  read: {
    id: `${scope}.read`,
    defaultMessage: 'Read',
  },
  reading: {
    id: `${scope}.reading`,
    defaultMessage: 'Reading',
  },
  confused: {
    id: `${scope}.confused`,
    defaultMessage: 'Confused',
  },
  doMagic: {
    id: `${scope}.doMagic`,
    defaultMessage: 'Do magic',
  },
  getAttention: {
    id: `${scope}.getAttention`,
    defaultMessage: 'Get attention',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
  suggest: {
    id: `${scope}.suggest`,
    defaultMessage: 'Suggest',
  },
  surprised: {
    id: `${scope}.surprised`,
    defaultMessage: 'Surprised',
  },
  congratulate: {
    id: `${scope}.congratulate`,
    defaultMessage: 'Congratulate',
  },
  flyIn: {
    id: `${scope}.flyIn`,
    defaultMessage: 'Fly in',
  },
  flyOut: {
    id: `${scope}.flyOut`,
    defaultMessage: 'Fly out',
  },
  standStill: {
    id: `${scope}.standStill`,
    defaultMessage: 'Stand still',
  },
  process: {
    id: `${scope}.process`,
    defaultMessage: 'Process',
  },
  listen: {
    id: `${scope}.listen`,
    defaultMessage: 'Listen',
  },
  think: {
    id: `${scope}.think`,
    defaultMessage: 'Think',
  },
  write: {
    id: `${scope}.write`,
    defaultMessage: 'Write',
  },
  idle: {
    id: `${scope}.idle`,
    defaultMessage: 'Idle',
  },
  turnPage: {
    id: `${scope}.turnPage`,
    defaultMessage: 'Turn page',
  },
  openBook: {
    id: `${scope}.openBook`,
    defaultMessage: 'Open book',
  },
  closeBook: {
    id: `${scope}.closeBook`,
    defaultMessage: 'Close book',
  },
  announce: {
    id: `${scope}.announce`,
    defaultMessage: 'Announce',
  },
  blink: {
    id: `${scope}.blink`,
    defaultMessage: 'Blink',
  },
  acknowledge: {
    id: `${scope}.acknowledge`,
    defaultMessage: 'Acknowledge',
  },
  acknowledgeThoughtful: {
    id: `${scope}.acknowledgeThoughtful`,
    defaultMessage: 'Acknowledge thoughtful',
  },
  browsUp: {
    id: `${scope}.browsUp`,
    defaultMessage: 'Brows up',
  },
  decline: {
    id: `${scope}.decline`,
    defaultMessage: 'Decline',
  },
  declineAnnoyed: {
    id: `${scope}.declineAnnoyed`,
    defaultMessage: 'Decline annoyed',
  },
  declineThoughtful: {
    id: `${scope}.declineThoughtful`,
    defaultMessage: 'Decline thoughtful',
  },
  pleased: {
    id: `${scope}.pleased`,
    defaultMessage: 'Pleased',
  },
  sad: {
    id: `${scope}.sad`,
    defaultMessage: 'Sad',
  },
  hearLeftEar: {
    id: `${scope}.hearLeftEar`,
    defaultMessage: 'Hear left ear',
  },
  hearRightEar: {
    id: `${scope}.hearRightEar`,
    defaultMessage: 'Hear right ear',
  },
  hearBothEars: {
    id: `${scope}.hearBothEars`,
    defaultMessage: 'Hear both ears',
  },
  yawn: {
    id: `${scope}.yawn`,
    defaultMessage: 'Yawn',
  },
  eatCracker: {
    id: `${scope}.eatCracker`,
    defaultMessage: 'Eat cracker',
  },
  wearSunglasses: {
    id: `${scope}.wearSunglasses`,
    defaultMessage: 'Wear sunglasses',
  },
  lookDown: {
    id: `${scope}.lookDown`,
    defaultMessage: 'Look down',
  },
  lookDownAndBlink: {
    id: `${scope}.lookDownAndBlink`,
    defaultMessage: 'Look down and blink',
  },
  glanceUp: {
    id: `${scope}.glanceUp`,
    defaultMessage: 'Glance up',
  },
  glanceDown: {
    id: `${scope}.glanceDown`,
    defaultMessage: 'Glance down',
  },
  glanceLeft: {
    id: `${scope}.glanceLeft`,
    defaultMessage: 'Glance left',
  },
  glanceRight: {
    id: `${scope}.glanceRight`,
    defaultMessage: 'Glance right',
  },
  moveLeft: {
    id: `${scope}.moveLeft`,
    defaultMessage: 'Move left',
  },
  moveRight: {
    id: `${scope}.moveRight`,
    defaultMessage: 'Move right',
  },
});
