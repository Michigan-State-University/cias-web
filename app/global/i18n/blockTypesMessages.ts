import { defineMessages } from 'react-intl';

export const scope = 'app.global.BlockTypes';

export default defineMessages({
  BodyAnimation: {
    id: `${scope}.BodyAnimation`,
    defaultMessage: 'Body Animation',
  },
  Speech: {
    id: `${scope}.Speech`,
    defaultMessage: 'Speech',
  },
  Reflection: {
    id: `${scope}.Reflection`,
    defaultMessage: 'Speech',
  },
  ReflectionFormula: {
    id: `${scope}.ReflectionFormula`,
    defaultMessage: 'Speech',
  },
  HeadAnimation: {
    id: `${scope}.HeadAnimation`,
    defaultMessage: 'Head Animation',
  },
  ReadQuestion: {
    id: `${scope}.ReadQuestion`,
    defaultMessage: 'Read question',
  },
  Pause: {
    id: `${scope}.Pause`,
    defaultMessage: 'Pause',
  },
  Feedback: {
    id: `${scope}.Feedback`,
    defaultMessage: 'Show Spectrum',
  },
});
