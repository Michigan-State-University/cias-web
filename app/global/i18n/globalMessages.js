import { defineMessages } from 'react-intl';

export const scope = 'app.GlobalMessages';

export default defineMessages({
  variables: {
    variableNamePlaceholder: {
      id: `${scope}.variableNamePlaceholder`,
      defaultMessage: 'Variable name...',
    },
    variableScorePlaceholder: {
      id: `${scope}.variableScorePlaceholder`,
      defaultMessage: 'Score',
    },
    value: {
      id: `${scope}.value`,
      defaultMessage: 'Value:',
    },
    emptyBadge: {
      id: `${scope}.emptyBadge`,
      defaultMessage: 'No assigned value',
    },
  },
  blockTypes: {
    BodyAnimation: {
      id: `${scope}.BodyAnimation`,
      defaultMessage: 'Body Animation',
    },
    Speech: {
      id: `${scope}.Speech`,
      defaultMessage: 'Speech',
    },
    Reflection: {
      id: `${scope}.Reflection`,
      defaultMessage: 'Speech',
    },
    HeadAnimation: {
      id: `${scope}.HeadAnimation`,
      defaultMessage: 'Head Animation',
    },
    ReadQuestion: {
      id: `${scope}.ReadQuestion`,
      defaultMessage: 'Read question',
    },
  },
  questionTypes: {
    'Question::Single': {
      id: `${scope}.Question::Single`,
      defaultMessage: 'Single answer screen',
    },
    'Question::Multiple': {
      id: `${scope}.Question::Multiple`,
      defaultMessage: 'Multi answer screen',
    },
    'Question::TextBox': {
      id: `${scope}.Question::TextBox`,
      defaultMessage: 'Textbox screen',
    },
    'Question::Number': {
      id: `${scope}.Question::Number`,
      defaultMessage: 'Number screen',
    },
    'Question::Grid': {
      id: `${scope}.Question::Grid`,
      defaultMessage: 'Grid screen',
    },
    'Question::AnalogueScale': {
      id: `${scope}.Question::AnalogueScale`,
      defaultMessage: 'Visual Analogue Scale screen',
    },
    'Question::Information': {
      id: `${scope}.Question::Information`,
      defaultMessage: 'Information screen',
    },
    'Question::Url': {
      id: `${scope}.Question::Url`,
      defaultMessage: 'Url screen',
    },
    'Question::FollowUpContact': {
      id: `${scope}.Question::FollowUpContact`,
      defaultMessage: 'FollowUpContact screen',
    },
  },
  statuses: {
    draft: {
      id: `${scope}.draft`,
      defaultMessage: 'Draft',
    },
    published: {
      id: `${scope}.published`,
      defaultMessage: 'Published',
    },
    closed: {
      id: `${scope}.closed`,
      defaultMessage: 'Closed',
    },
    archived: {
      id: `${scope}.archived`,
      defaultMessage: 'Archived',
    },
  },
  createProblemError: {
    id: `${scope}.createProblemError`,
    defaultMessage: `Couldn't create an intervention`,
  },
  editProblemError: {
    id: `${scope}.editProblemError`,
    defaultMessage: `Couldn't edit an intervention`,
  },
});
