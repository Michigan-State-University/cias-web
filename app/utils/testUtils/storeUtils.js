import { createStore } from 'redux';
import createReducer from 'reducers';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

export const createTestStore = (initialState) => {
  const reducer = (state) => state;

  const store = createStore(reducer, {
    auth: authReducerInitialState,
    ...initialState,
  });

  store.runSaga = () => {};
  store.injectedReducers = {};
  store.injectedSagas = {};
  store.createReducer = createReducer;

  return store;
};
