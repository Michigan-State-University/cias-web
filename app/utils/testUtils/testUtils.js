import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { DEFAULT_LOCALE } from 'i18n';

const ProviderWrapper = ({ children }) => (
  <IntlProvider locale={DEFAULT_LOCALE}>{children}</IntlProvider>
);

const testRender = (ui, options) =>
  render(ui, { wrapper: ProviderWrapper, ...options });

export { testRender };

ProviderWrapper.propTypes = {
  children: PropTypes.node,
};
