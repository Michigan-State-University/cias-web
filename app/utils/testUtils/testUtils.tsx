import React, { ComponentType, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { DEFAULT_LOCALE } from 'i18n';

type Props = { children: JSX.Element };

const ProviderWrapper = ({ children }: Props): JSX.Element => (
  <IntlProvider locale={DEFAULT_LOCALE}>{children}</IntlProvider>
);

const testRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult =>
  render(ui, { wrapper: ProviderWrapper as ComponentType, ...options });

export { testRender };
