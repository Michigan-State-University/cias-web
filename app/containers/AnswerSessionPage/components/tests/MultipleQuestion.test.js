/**
 *
 * Tests for MultipleQuestion
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';
import { DEFAULT_LOCALE } from 'i18n';

import { createTestStore } from 'utils/testUtils/storeUtils';
import { Provider } from 'react-redux';
import MultipleQuestion from '../MultipleQuestion';

describe('<MultipleQuestion />', () => {
  const store = createTestStore({});

  const mockedFunctions = {
    selectAnswer: jest.fn(),
  };

  const defaultProps = {
    question: {
      body: {
        data: [
          {
            payload: 'test',
            variable: { name: '1', value: '1' },
          },
          {
            payload: 'test',
            variable: { name: '2', value: '2' },
          },
        ],
      },
      id: 'test-id',
    },
    answerBody: [],
    ...mockedFunctions,
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter>
            <MultipleQuestion {...defaultProps} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter>
            <MultipleQuestion {...defaultProps} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
