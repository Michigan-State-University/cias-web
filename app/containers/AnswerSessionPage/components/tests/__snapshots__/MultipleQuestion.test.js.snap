// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleQuestion /> Should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  margin-bottom: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 44px;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 10px;
  padding-bottom: 10px;
  height: 44;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: none;
}

.c10 {
  margin-right: 5px;
}

.c5:disabled + label > .c9 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c8:hover {
  cursor: pointer;
}

.c5:disabled + .c7 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c2 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: calc(100% + 21px);
  padding-top: 14px;
  padding-bottom: 14px;
  padding-right: 21px;
  padding-left: 21px;
  cursor: pointer;
  border-radius: 5px;
  border-radius: 5px;
}

.c2:hover {
  background-color: #E9F1FB;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="HoverableBox-sc-zvip6a-0 c2"
        width="calc(100% + 21px)"
      >
        <div
          class="c3"
          height="44"
        >
          <div
            class="c4"
          >
            <input
              class="c5 c6"
              id="answer-1"
              type="checkbox"
            />
            <label
              class="c7 c8"
              for="answer-1"
            >
              <img
                class="c9 c10"
                src="IMAGE_MOCK"
              />
              <div>
                test
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="HoverableBox-sc-zvip6a-0 c2"
        width="calc(100% + 21px)"
      >
        <div
          class="c3"
          height="44"
        >
          <div
            class="c4"
          >
            <input
              class="c5 c6"
              id="answer-2"
              type="checkbox"
            />
            <label
              class="c7 c8"
              for="answer-2"
            >
              <img
                class="c9 c10"
                src="IMAGE_MOCK"
              />
              <div>
                test
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
